platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/console/UserInputReceiver.java

```diff
@@ -24,5 +26,15 @@ public interface UserInputReceiver {
      * Requests that a line of text should be received from the user, for example via this process' stdin, and forwarded to the daemon.
      * Does not block waiting for the input.
      */
-    void readAndForwardText();
+    void readAndForwardText(Normalizer normalizer);
```

Comment:
Both the implementations use the `normalizer` to validate and normalize the text prior to forwarding it to the daemon. The implementations can't do their job without it. The validation and normalization are also both specific to the current request. Both of these things mean it makes sense to pass this as a parameter, rather than, say, injecting it into a constructor.